extensions/amp-ad-network-fake-impl/0.1/amp-ad-network-fake-impl.js

```diff
@@ -59,15 +59,24 @@ export class AmpAdNetworkFakeImpl extends AmpA4A {
           // and the signature is "FAKESIG". This mode is only allowed in
           // `localDev` and primarily used for A4A Envelope for testing.
           // See DEVELOPING.md for more info.
-          const creative = this.transformCreativeLocalDev_(deserialized);
+          let creative = this.transformCreativeLocalDev_(deserialized);
+          try {
+            // Gracefully handle the case in which the fake ad is still
+            // formatted in JSON, which it almost certainly is, since
+            // "gulp serve" won't serve an HTML file under /extensions
+            const json = JSON.parse(deserialized);
```

Comment:
Basically the issue here is that normally it wants JSON like { "creative": ...html..., "signature": ... }
However if you use a fake ad, then it was just rendering everything it gets, so the ad would contain the JSON structure above, not just the html. But if you change <amp-ad src=foo.json> to <amp-ad src=foo.html>, then it doesn't work because gulp won't serve a .html file from extensions/. So the ugly workaround was to make a .json file which contained no JSON, only HTML. This fix just solves that, and affects only the fake ad network.
A real ad would come in the JSON format, but we would never be in AmpAdNetworkFakeImpl. extractCreativeAndSignature() in the first place.